core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java

```diff
@@ -155,20 +159,22 @@ private void configureAndLaunchPlugins(
 
 	private ConsumerConnector buildKafkaConsumer() {
 
+		Properties kafkaProperties = kafkaOptions.getProperties();
+
 		final ConsumerConnector consumer = Consumer.createJavaConsumerConnector(new ConsumerConfig(
-				kafkaOptions.getProperties()));
+				kafkaProperties));
 
 		return consumer;
 	}
 
-	private void launchTopicConsumer(
+	private Future<?> launchTopicConsumer(
 			final String formatPluginName,
 			final AvroFormatPlugin<?, ?> avroFormatPlugin,
 			final KafkaIngestRunData ingestRunData,
 			final List<String> queue )
 			throws Exception {
 		final ExecutorService executorService = Executors.newFixedThreadPool(queue.size());
```

Comment:
I just realized something here.. Is this creating a thread pool of X size for every single topic?  Like if I had 3 topics it would create 3 threadpools of 3 max threads each?  I think it would probably only actually create 3 threads but there might be overhead from the extra pools...